### PR TITLE
Added required condition to server room in DC view

### PR DESCRIPTION
**Pull request by ar4s**
Server room must be specify when user add rack from DC view.
Connected to #1799

### CODE REVIEW

File: src/ralph/dc_view/static/partials/data_center/data_center.html [hunk 1]
```diff
@@ -46,13 +46,15 @@
                   </div>
                 </div>
                 <div class="row">
-                  <div class="large-12 columns">
+                  <div class="large-12 columns" ng-class="{'error': forms.edit_form.server_room.$invalid}">
                       <label for="id_name">Server room
-                        <select name="server_room" ng-model="rack.server_room"  required>
+                        <select name="server_room" ng-model="rack.server_room" required>
+                          <option value="">-----</option>
```
Comment (mkurek) [inline src/ralph/dc_view/static/partials/data_center/data_center.html]: if this field is required why do you add empty option?

File: src/ralph/dc_view/static/js/app/data_center/controllers.js [hunk 2]
```diff
@@ -28,6 +29,20 @@ angular
...
         };
+=======
+            $scope.newRack = function(data_center, x, y) {
+                var rack = new RackModel();
+                rack.visualization_col = x;
+                rack.visualization_row = y;
+                rack.orientation = 'top';
+                rack.server_room = '';
+                rack.name = 'New rack';
+                rack.data_center = data_center.id;
```
Comment (mkurek) [inline src/ralph/dc_view/static/js/app/data_center/controllers.js]: in ng model rack is not connected to dc directly, but through server room

File: src/ralph/dc_view/static/partials/data_center/data_center.html (hunk 1 above)
Comment (ar4s) [inline src/ralph/dc_view/static/partials/data_center/data_center.html]: ask angular
mind_blow

File: src/ralph/dc_view/static/js/app/data_center/controllers.js (hunk 2 above)
Comment (ar4s) [inline src/ralph/dc_view/static/js/app/data_center/controllers.js]: you right.. there are some legacy code from 2.x

**Comment by mkurek** [comment]
:ok:

**Comment by ghost** [comment]
:ok:

**Pull request merged**